dallinger/command_line/ec2.py

```diff
@@ -0,0 +1,233 @@
+        config.load()
+    return {
+        "pem": config.get("ec2_default_pem", "dallinger"),
+        "security_group_name": config.get("ec2_default_security_group", "dallinger"),
+    }
+
+
+# EC2 on demand provisioning
+@click.group("ec2")
+@click.pass_context
+def ec2(ctx):
+    pass
+
+
+@ec2.group("ssh")
+@click.pass_context
+def ssh(ctx):
+    pass
+
+
+@ssh.command("web")
+@click.option("--app", required=True, help="App name")
+@click.option("--dns", required=True, help="Server name")
+def ssh__web(app, dns):
+    command = f"ssh {dns} -t 'docker exec -it {app}-web-1 bash'"
+    os.system(command)
+
+
+@ec2.group("list")
+@click.pass_context
+def list(ctx):
+    pass
+
+
+@list.command("instances")
+@click.option("--region", default=None, help="Region name")
+@click.option("--running", is_flag=True, help="List running instances")
+@click.option("--stopped", is_flag=True, help="List stopped instances")
+@click.option("--terminated", is_flag=True, help="List terminated instances")
+@click.option("--pem", default=None, help="Name of the PEM file to use")
+@click.pass_context
+def list__instances(ctx, region, running, stopped, terminated, pem):
+    filtered_states = []
+    if running:
+        filtered_states.append("running")
+    if stopped:
+        filtered_states.append("stopped")
+    if terminated:
+        filtered_states.append("terminated")
+    list_instances(region, filtered_states=filtered_states, pem=pem)
+
+
+@list.command("regions")
+@click.pass_context
+def list__regions(ctx):
+    list_regions()
+
+
+@list.command("instance_types")
+@click.option("--region", default=None, help="Region name")
+@click.pass_context
+def list__instance_types(ctx, region):
+    list_instance_types(region)
+
+
+@ec2.command("provision")
+@click.option("--name", required=True, help="Instance name")
+@click.option("--region", default=None, help="Region name")
+@click.option("--type", default="m5.xlarge", help="Instance type")
+@click.option("--storage", default=32, type=int, help="Storage in GB; default is 32 GB")
+@click.option(
+    "--pem", default="dallinger", help="PEM file name; default is dallinger.pem"
+)
+@click.option(
+    "--image_name",
+    default="ubuntu/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-20230516",
+    help="Image name; default is Ubuntu 22.04",
+)
+@click.option(
+    "--security_group_name",
+    default="dallinger",
+    help="Security group name; default is dallinger",
+)
+@click.option(
+    "--dns-host",
+    help="DNS name to use. Must resolve all its subdomains to the IP address specified as ssh host",
+    default=None,
+)
+@click.pass_context
+def ec2__provision(
+    ctx, name, region, type, storage, pem, image_name, security_group_name, dns_host
+):
+    try:
+        from dallinger import db, experiment
+
+        config = get_instance_config()
+        exp_klass = experiment.load()
+        exp = exp_klass(db.session)
+
+        asset_storage = getattr(exp, "asset_storage", None)
```

Comment:
This logic about asset storage can be removed; not only is it PsyNet-specific, but it also is out of date.

Reply:
I was wondering about that. I tried to generalize and decouple it from PsyNet as much as possible, but it's great to hear it can be removed.